app/controllers/v3/auth/federation_controller.rb

```diff
@@ -1,9 +1,61 @@
 module V3
   module Auth
     class FederationController < ApplicationController
-      def oidc; end
+      TIME_FORMAT = '%Y-%m-%dT%H:%M:%S.%L000Z'.freeze
 
-      def voms; end
+      def oidc
+        credentials = ::Auth::Oidc.unified_credentials(ENV.select { |name| name.start_with?('OIDC') })
+        headers['X-Subject-Token'] = ::Tokenator.to_token(credentials.to_hash)
```

Comment:
How are encryption/decryption errors handled? I don't see any error handling code in `Tokenator`. Specially for decryption ... "users" will attempt to send you all sorts of shit, dropping out with unhandled HTTP[500] might not be the best choice here.